microprofile-metrics/src/test/java/org/jboss/eap/qe/microprofile/metrics/integration/config/CustomMetricBaseTest.java

```diff
@@ -0,0 +1,93 @@
+package org.jboss.eap.qe.microprofile.metrics.integration.config;
+
+import static io.restassured.RestAssured.get;
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.equalTo;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SUBSYSTEM;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.concurrent.TimeoutException;
+
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.as.arquillian.api.ContainerResource;
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.ConfigurationException;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.creaper.ManagementClientProvider;
+import org.junit.Before;
+import org.junit.Test;
+import org.wildfly.extras.creaper.core.online.operations.admin.Administration;
+
+import io.restassured.http.ContentType;
+
+public abstract class CustomMetricBaseTest {
+
+    public static final String INCREMENT_CONFIG_PROPERTY = "dummy.increment";
+
+    public static final int DEFAULT_VALUE = 2;
+
+    private static final PathAddress CONFIG_SOURCE_PROPS_ADDRESS = PathAddress.pathAddress()
+            .append(SUBSYSTEM, "microprofile-config-smallrye")
+            .append("config-source", "props");
+
+    @ArquillianResource
+    URL deploymentUrl;
+
+    @ContainerResource
+    ManagementClient managementClient;
+
+    String metricsURL;
+
+    @Before
+    public void before() throws ConfigurationException, InterruptedException, TimeoutException, IOException {
+        metricsURL = "http://" + managementClient.getMgmtAddress() + ":" + managementClient.getMgmtPort() + "/metrics";
+        new Administration(ManagementClientProvider.onlineStandalone()).reload();
+    }
+
+    abstract void setConfigProperties(int number) throws Exception;
+
+    /**
+     * @tpTestDetails Multi-component customer scenario to verify correct behaviour when custom counter metric is used.
+     *                The metric increment depends on CDI bean and MP Config property value.
+     * @tpPassCrit Counter metric has correct value
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test
+    public void testCustomMetricDefault() throws Exception {
+        setConfigProperties(DEFAULT_VALUE);
+        performRequest();
+        testMetricForValue(DEFAULT_VALUE);
+    }
+
+    /**
+     * @tpTestDetails High level customer scenario to verify correct behaviour when custom counter metric is used.
+     *                The metric increment depends on CDI bean and MP Config property value.
+     * @tpPassCrit Counter metric has correct value
```

Comment:
What is correct value?

Reply:
Such detail does not need to be in test plan.

Reply:
I should be able to determine from this documentation what exactly this test do. Every test is testing, that something has either bad or correct value. IMHO it is important to mention _what_ is a "correct" value in this context.

Reply:
MP Metric Counter is counting invocations. Specific number should not be here.

Reply:
I've never mentioned that there should be specific value mentioned. The important thing is what "correct" value represents in this case. Such as: `@tpPassCrit Counter metric value equals to count of invocations of <something>`

Reply:
Would you be OK with `Counter metric is incremented by configured value`?

Reply:
Thanks! It is better! I would still want to know when/how the "counter metric" is incremented and where the "configured value" came from ("...CDI bean and MP Config property value." - is it first or second?).

Reply:
depends on MP Config and is provided by a CDI bean
I have updated the doc.